src/Modules/Guilds/pages/AllProposals/AllProposals.tsx

```diff
@@ -0,0 +1,100 @@
+import { useGuildProposalIds } from 'hooks/Guilds/ether-swr/guild/useGuildProposalIds';
+import { Filter } from 'Components/Filter';
+import ProposalCardWrapper from 'Modules/Guilds/Wrappers/ProposalCardWrapper';
+import { GuildAvailabilityContext } from 'contexts/Guilds/guildAvailability';
+import Result, { ResultState } from 'old-components/Guilds/common/Result';
+import { useContext, useEffect, useMemo, useState } from 'react';
+import { Virtuoso } from 'react-virtuoso';
+import { useFilter } from 'contexts/Guilds';
+import { ProposalListWrapper, ProposalsList } from './AllProposals.styled';
+
+const AllProposals = ({ guildId }) => {
+  const { isLoading } = useContext(GuildAvailabilityContext);
+  const { data: proposalIds, error } = useGuildProposalIds(guildId);
+  const [openSearchBar, setOpenSearchBar] = useState(false);
+
+  /*
+  In "Governance" view, we reset all filters and apply one to show
+  only "Active" proposals. Once we switch to "All proposals" view, 
+  we reset the "Active" proposals filter. Also reset the "Action" and
+  "Currency" filters to be consistant.
+  */
+
+  const { onResetState, onResetActionType, onResetCurrency, setSearchQuery } =
+    useFilter();
+  // Reset filters when page loads
+  useEffect(() => {
+    setSearchQuery('');
+    onResetActionType();
+    onResetCurrency();
+    onResetState();
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
+  const revertedProposals = useMemo(() => {
+    if (!proposalIds) return null;
+    // clone array as the original proposalIds array from Ethers is immutable
+    const clone = [...proposalIds];
+    // Show latest proposals first
+    return clone.reverse();
+  }, [proposalIds]);
+
+  const PROPOSALS_TO_LOAD = 10;

```

Comment:
Can you define this constant outside your FunctionalComponent?

Reply:
I can, but where will it make sense?

I extracted a constant as a way of "naming" an arbitrary value of number of proposals to load. This as well could be 5, 15 or 20. But this is only relevant in this context.

Reply:
Outside your component definition is ok for now I think. Just to avoid creating the same const on each render.